tests/unit/gateways/clients/test_elastic_search_client.py

```diff
@@ -0,0 +1,116 @@
+import pytest
+
+from gateways.clients.elastic_search_client import ElasticSearchClient
+
+
+class TestElasticSearchClient:
+
+    def test_empty_search_string(self):
+        client = ElasticSearchClient()
+        search_text = ""
+        sort_by = ""
+        order = ""
+        search_after = []
+
+        result = client._build_search_query(search_text, sort_by, order, search_after)
+
+        with pytest.raises(KeyError):
+            result['query']
+
+        with pytest.raises(KeyError):
+            result['search_after']
+
+    def test_filled_search_string(self):
+        client = ElasticSearchClient()
+        search_text = "Test"
+        sort_by = ""
+        order = ""
+        search_after = []
+
+        result = client._build_search_query(search_text, sort_by, order, search_after)
+        assert result['query'] is not None
+        assert result['query']['multi_match'] is not None
+        assert result['query']['multi_match']['query'] == search_text
+
+    def test_search_query_without_search_after(self):
+        client = ElasticSearchClient()
+        search_text = "Test"
+        sort_by = "name"
+        order = "asc"
+        search_after = []
+
+        result = client._build_search_query(search_text, sort_by, order, search_after)
+
+        with pytest.raises(KeyError):
+            result['search_after']
```

Comment:
It's not much different, but this is more readable
```suggestion
        assert 'search_after' not in result
```

Reply:
Great catch. I've changed that in the new commit